decode.go

```diff
@@ -18,12 +18,6 @@ const (
 	disallowUnknownFieldsFlag
 )
 
-const (
-	bytesAllocLimit = 1e6 // 1mb
-	sliceAllocLimit = 1e4
-	maxMapSize      = 1e6
-)
```

Comment:
Without this it is trivial to construct a tiny malicious MsgPack message that will try to allocate 1TB and fail with a panic. That is unacceptable for public API.

We should either
- keep this under a flag
- or recover from panic

What do you think?:

Reply:
...and recovering from a panic won't help much when malicious user tries to allocate 1gb instead of 1tb.